crates/router/src/connector/adyen/transformers.rs

```diff
@@ -566,98 +611,139 @@ fn get_country_code(item: &types::PaymentsAuthorizeRouterData) -> Option<api_enu
 fn get_payment_method_data<'a>(
     item: &types::PaymentsAuthorizeRouterData,
 ) -> Result<AdyenPaymentMethod<'a>, error_stack::Report<errors::ConnectorError>> {
-    match item.request.payment_method_data {
-        api::PaymentMethodData::Card(ref card) => {
-            let adyen_card = AdyenCard {
+    match item
+        .request
+        .mandate_id
+        .to_owned()
+        .and_then(|mandate_ids| mandate_ids.mandate_reference_id)
+    {
+        Some(api_models::payments::MandateReferenceId::ConnectorMandateId(
+            connector_mandate_id,
+        )) => {
+            let adyen_mandate = AdyenMandate {
                 payment_type: PaymentType::Scheme,
-                number: card.card_number.clone(),
-                expiry_month: card.card_exp_month.clone(),
-                expiry_year: card.card_exp_year.clone(),
-                cvc: Some(card.card_cvc.clone()),
+                stored_payment_method_id: connector_mandate_id,
             };
-            Ok(AdyenPaymentMethod::AdyenCard(adyen_card))
+            Ok(AdyenPaymentMethod::Mandate(adyen_mandate))
         }
-        api::PaymentMethodData::Wallet(ref wallet_data) => match wallet_data {
-            api_models::payments::WalletData::GooglePay(data) => {
-                let gpay_data = AdyenGPay {
-                    payment_type: PaymentType::Googlepay,
-                    google_pay_token: data.tokenization_data.token.to_owned(),
-                };
-                Ok(AdyenPaymentMethod::Gpay(gpay_data))
-            }
-            api_models::payments::WalletData::ApplePay(data) => {
-                let apple_pay_data = AdyenApplePay {
-                    payment_type: PaymentType::Applepay,
-                    apple_pay_token: data.payment_data.to_string(),
-                };
-
-                Ok(AdyenPaymentMethod::ApplePay(apple_pay_data))
-            }
-            api_models::payments::WalletData::PaypalSdk(_) => {
-                Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into())
+        Some(api_models::payments::MandateReferenceId::NetworkMandateId(network_mandate_id)) => {
+            match item.request.payment_method_data {
+                api::PaymentMethodData::Card(ref card) => {
+                    let adyen_card = AdyenCard {
+                        payment_type: PaymentType::Scheme,
+                        number: card.card_number.clone(),
+                        expiry_month: card.card_exp_month.clone(),
+                        expiry_year: card.card_exp_year.clone(),
+                        cvc: None,
+                        brand: Some(CardBrand::Visa), //Fixme
```

Comment:
Explain the problem for FIXME. If there is a issue already linked add the issue here.

Reply:
handled